### PR TITLE
107496 refactor get by guid async

### PR DESCRIPTION
**Pull request by janid1967**
[AB#107496](https://statoildeveloper.visualstudio.com/b6d97460-2e53-47ae-aad4-b41396d90828/_workitems/edit/107496) 

Will rename IRepositoryWithGuid.GetByGuidAsync to GetAsync after approval. The rename make a lot of "noise" unnecessary to review

### CODE REVIEW

File: src/Equinor.ProCoSys.Completion.Infrastructure/Repositories/EntityWithGuidRepository.cs [hunk 1]
```diff
@@ -18,6 +19,17 @@ protected EntityWithGuidRepository(CompletionContext context, DbSet<TEntity> set
...
+        if (entity is null)
+        {
+            var type = typeof(TEntity).Name;
+            throw new EntityNotFoundException($"Could not find {type} with Guid {guid}");
+        }
+        return entity;
+    }
+
+    public virtual async Task<bool> ExistsAsync(Guid guid)
+        => await Set.AnyAsync(e => e.Guid == guid);
```
Comment (jsoi-eq) [inline src/Equinor.ProCoSys.Completion.Infrastructure/Repositories/EntityWithGuidRepository.cs:34]: Curious: Why is Set used instead of DefaultQuery?

Comment (janid1967) [reply src/Equinor.ProCoSys.Completion.Infrastructure/Repositories/EntityWithGuidRepository.cs:34]: Speed! The DefaultQuery can have a lot of .Include (which create Joins), which who don' need here

File: src/tests/Equinor.ProCoSys.Completion.Infrastructure.Tests/EntityWithGuidRepositoryTestBase.cs [hunk 2]
```diff
@@ -101,10 +102,23 @@ public async Task GetByGuid_KnownGuid_ShouldReturnEntity()
     }
 
     [TestMethod]
-    public async Task GetByGuid_UnknownGuid_ShouldReturnNull()
+    public async Task GetByGuid_UnknownGuid_ShouldThrowEntityNotFoundException()
+        // Act and Assert
+        => await Assert.ThrowsExceptionAsync<EntityNotFoundException>(() => _dut.GetByGuidAsync(Guid.Empty));
```
Comment (jsoi-eq) [inline src/tests/Equinor.ProCoSys.Completion.Infrastructure.Tests/EntityWithGuidRepositoryTestBase.cs:107]: Are you able to assert that we get the entity name and the guid in the exception message as well?

**Review by jsoi-eq — APPROVED** [review]

**Comment by jsoi-eq** [comment]
Very happy that we get rid of a lot of null-forgiving operators.

**Pull request merged**